src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,97 @@
-        return "Today";
+        switch (datePart) {
+            case FULL:
+                return LocalDate.now().toString();
+            case YEAR:
+                return String.valueOf(LocalDate.now().getYear());
+            case MONTH:
+                return LocalDate.now().getMonth().toString();
+            case DAY:
+                return String.valueOf(LocalDate.now().getDayOfMonth());
+            default:
+                throw new DateTimeException("No such option");
+        }
     }
-
-    /**
-     * Given an Array of 3 elements, where
-     *         - 1-st element is a `year`;
-     *         - 2-nd element is s `month`;
-     *         - 3-rd element is a `day of month`;
-     *
-     * Return Optional of a date built from these elements.
-     */
+    
     public Optional<LocalDate> getDate(Integer[] dateParams) {
-        return Optional.empty();
+        int year = 0;
+        int month = 1;
+        int day = 2;
+        if (dateParams.length == 0 || dateParams[month] > 12 || dateParams[day] > 31) {
+            return Optional.empty();
+        }
+        return Optional.of(LocalDate.of(dateParams[year], dateParams[month], dateParams[day]));
     }
-
-    /**
-     * Given the time and the number of hours to add, return the changed time.
-     */
+    
     public LocalTime addHours(LocalTime localTime, Integer hoursToAdd) {
-        return LocalTime.now();
+        return localTime.plusHours(hoursToAdd);
     }
-
-    /**
-     * Given the time and the number of minutes to add, return the changed time.
-     */
+    
     public LocalTime addMinutes(LocalTime localTime, Integer minutesToAdd) {
-        return LocalTime.now();
+        return localTime.plusMinutes(minutesToAdd);
     }
-
-    /**
-     * Given the time and the number of seconds to add, return the changed time.
-     */
+    
     public LocalTime addSeconds(LocalTime localTime, Integer secondsToAdd) {
-        return LocalTime.now();
+        return localTime.plusSeconds(secondsToAdd);
     }
-
-    /**
-     * Given the date and the number of weeks to add, return the changed date.
-     */
+    
     public LocalDate addWeeks(LocalDate localDate, Integer numberOfWeeks) {
-        return LocalDate.now();
+        return localDate.plusWeeks(numberOfWeeks);
     }
-
-    /**
-     * Given a random `someDate` date, return one of the following Strings:
-     *     - "`someDate` is after `currentDate`"
-     *                  if `someDate` is in the future relating to the `current date`;
-     *     - "`someDate` is before `currentDate`"
-     *                  if `someDate` is in the past relating to the `current date`;
-     *     - "`someDate` is today"
-     *                  if `someDate` is today;
-     */
+    
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        if (someDate.isAfter(LocalDate.now())) {
+            return someDate.toString() + " is after " + LocalDate.now().toString();
```

Comment:
```suggestion
            return someDate + " is after " + LocalDate.now();
```
since you use concat here, toString() is not necessary

Reply:
Thanks)